### PR TITLE
TIP-713: Allow variant group values to be empty

### PR DESCRIPTION
**Pull request by damien-carcel**
## Description

When adding an attribute on the variant group, value can be set empty. Expected behavior is to have all products attached to the variant group with this value empty.

Previous solution was a patapouille where a temporary product is created, updated with the previous variant group values, then the new ones. Finally, this temporaty product ProductValueCollection is set on the variant.

Problem is empty values were ignored. So the VariantGroupUpdater don't use a temporary product anymore, but directly the ProductValueFactory. New values are created, then added (or replace already existing ones) to the template value collection.

This new way implies to take care of media in the VariantGroupUpdater, as their creation is the reponsibility of the updaters (so in the previous way, of the ProductUpdater that was used), not of the ProductValueFactory.

## Bonus

I made some cleanup of stuff I encountered during the debugging phase: some useless `use` statements and missing phpdoc, but also the `AddParametersToVariantProductGridListener`. The service is still used and mandatory, but the class extended `AddParametersToProductGridListener` for nothing, as all the specific things it did before were deactivated and not used anymore since the product datagrid uses ES.

## Definition Of Done

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | OK
| Added Behats                      | -
| Added integration tests           | -
| Changelog updated                 | OK
| Review and 2 GTM                  | OK
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Pim/Bundle/DataGridBundle/Datasource/Datasource.php [hunk 1]
```diff
@@ -147,10 +148,6 @@ public function setParameters($parameters)
         if ($this->qb instanceof QueryBuilder) {
             $this->qb->setParameters($this->parameters);
         }
-        // TODO - TIP-664: make the datagrid work with ES
-        elseif ($this->qb instanceof \Pim\Bundle\ElasticSearchBundle\Query\QueryBuilder) {
```
Comment (damien-carcel) [inline src/Pim/Bundle/DataGridBundle/Datasource/Datasource.php:151]: This class does not exists, so this `elseif` is never called.

**Review by jjanvier — APPROVED** [review]
:+1: GTM

File: src/Pim/Bundle/DataGridBundle/Extension/MassAction/MassActionDispatcher.php [hunk 2]
```diff
@@ -77,10 +77,8 @@ public function dispatch(Request $request)
         $parameters = $this->prepareMassActionParameters($request);
         $datagrid = $parameters['datagrid'];
         $massAction = $parameters['massAction'];
-        $inset = $parameters['inset'];
-        $values = $parameters['values'];
 
-        return $this->performMassAction($datagrid, $massAction, $inset, $values);
+        return $this->performMassAction($datagrid, $massAction);
```
Comment (momoss) [inline src/Pim/Bundle/DataGridBundle/Extension/MassAction/MassActionDispatcher.php:81]: :+1:

File: src/Pim/Bundle/DataGridBundle/EventListener/AddParametersToVariantProductGridListener.php [hunk 3]
```diff
@@ -1,70 +0,0 @@
-<?php
```
Comment (momoss) [inline src/Pim/Bundle/DataGridBundle/EventListener/AddParametersToVariantProductGridListener.php:1]: Are you sure, that we dont need this listener in the future?

File: src/Pim/Bundle/EnrichBundle/Resources/config/datagrid_listeners.yml [hunk 4]
```diff
@@ -62,13 +62,12 @@ services:
             - { name: kernel.event_listener, event: oro_datagrid.datgrid.build.before.product-variant-group-grid, method: buildBefore }
 
     pim_enrich.event_listener.product_variant_group_grid_after_listener:
-        class: '%pim_datagrid.event_listener.add_parameters_to_variant_product_grid.class%'
```
Comment (momoss) [inline src/Pim/Bundle/EnrichBundle/Resources/config/datagrid_listeners.yml:65]: Don't forget to remove class definition plz

**Review by momoss — APPROVED** [review]

**Review by jjanvier — APPROVED** [review]

File: src/Pim/Bundle/DataGridBundle/EventListener/AddParametersToVariantProductGridListener.php (hunk 3 above)
Comment (damien-carcel) [reply src/Pim/Bundle/DataGridBundle/EventListener/AddParametersToVariantProductGridListener.php:1]: The listener itself is still here as we need here. Only the class is removed as it does nothing.

File: src/Pim/Bundle/EnrichBundle/Resources/config/datagrid_listeners.yml (hunk 4 above)
Comment (damien-carcel) [reply src/Pim/Bundle/EnrichBundle/Resources/config/datagrid_listeners.yml:65]: Done

**Review by SamirBoulil — APPROVED** [review]

**Pull request merged**